lock-impl/src/main/java/com/palantir/lock/impl/LockThreadInfoSnapshotManager.java

```diff
@@ -65,10 +77,22 @@ private void run() {
         scheduleRun();
     }
 
-    private void scheduleRun() {
-        if (threadInfoConfiguration.recordThreadInfo()) {
+    @VisibleForTesting
+    synchronized boolean isScheduled() {
+        return isScheduled;
+    }
+
+    private synchronized void scheduleRun() {
+        if (threadInfoConfiguration.current().recordThreadInfo()) {
+            if (!isScheduled) {
```

Comment:
Instead of using Atomics, synchronizing all blocks that touch `isScheduled` appears easier. Synchronization overhead is negligible.